### ISSUE TITLE
Help: Dotnet not found on Steam Deck even though installed

### ISSUE THREAD

**Issue by mihiros6000** [Reloaded-Project/Reloaded-II#513]
**Did you install the Prerequisites?**
Yes, checked + running  the Installer for dotnet 9 windows desktop runtime x86/x64 shows Repair/Uninstall options.

**Describe the Issue**
Prerequisites installed + vcredist x86/x64 + SetupLinux run and I tried manual extraction just in case too. Running ReloadedII.exe via Metaphor (game in question) prefix always spits out:
![Image](https://github.com/user-attachments/assets/a72b3e7e-6650-4869-9ba3-eb5a6549b9a3)

For transparency, I was trying to get SpecialK working (gave up) which is when the troubles started. However, I have nuked and recreated the prefix multiple times and still hitting the issue. However, I thought it might be worth mentioning just in case SpecialK debris somehow got added to the compatdata/prefix creation requirements and are conflicting with dotnet.

**Additional Resources**
Game is Metaphor: ReFantazio
I had it working before, so I know it's possible lol

**Comment by Sewer56** [comment]
Copies from a comment I've made before.

====

What happens under Wine is.

1. Wine inherits the environment variables of the host system.
2. The .NET Runtime allows you to override its expected location with the `DOTNET_ROOT` environment variable.
3. When you have a native Linux installation of .NET, the package sets `DOTNET_ROOT` as many distros install the runtime in a non-default location.

This is problematic, because when you try to run a .NET application in Wine, it looks at the native Linux one, rather than the Windows one.

The `Setup-Linux.exe` installer sets an [environment variable](https://github.com/Reloaded-Project/Reloaded-II/blob/0037b18e212607c981c085d14aa6bcddc3f26b5b/source/Reloaded.Mod.Installer.Lib/MainWindowViewModel.cs#L115) to explicitly state `DOTNET_ROOT` to be in `Program Files/dotnet` to avoid it inheriting the value from the Linux host. I suspect something is borked around that.

----

So in any case, you can try running the `Exec` 
command from the `.desktop` file shortcut made by the installer that's on your Desktop. See what happens if you prefix it with `DOTNET_ROOT="" `, so for example `DOTNET_ROOT="" protontricks-launch ...`

**Comment by Sewer56** [comment]
You can also try running `dotnet --list-runtimes` via protontricks-launch, see what you get for your game's WINEPREFIX.

**Comment by mihiros6000** [comment]
That worked, but the same thing seems to be happening with vc_redist x86/x64
![Image](https://github.com/user-attachments/assets/e412c012-2305-4a91-9bf6-8a7f43c82d02)

**Comment by Sewer56** [comment]
These deps are usually installed as part of `Setup-Linux.exe`, I wonder if you wiped your WINEPREFIX after running it, hmm.

Just install into the prefix and you should be good.

**Comment by mihiros6000** [comment]
Unfortunately, that didn't seem to work. I installed them to the prefix and still got these messages. At this point, I'm currently running DOTNET_ROOT="" protontricks-launch Reloaded-II.exe

**Comment by mihiros6000** [comment]
The application seems to work, the check is just messed up probably. However, it seems like the workaround nature of how I'm running it has stopped me from seeing Z:/ drive when adding an application.
![Image](https://github.com/user-attachments/assets/ee345e42-0a6c-4d27-b3f2-c6c50d7a2a4b)

**Comment by Sewer56** [comment]
Note: VCRedist is required only by the loader, the launcher only uses .NET.

**Comment by mihiros6000** [comment]
Ah ok, so I'd have to check once I've added the game + some mods and attempt a load?

**Comment by Sewer56** [comment]
Basically you'll know simply by whether the debug/log console shows up, yeah.

**Comment by mihiros6000** [comment]
Do you know why I might not have access to Z:/ and files to get to the game's exe?

**Comment by Sewer56** [comment]
No, it's highly unusual, I haven't seen a case like that before.
Technically you can manually map it with winecfg, but it's strange in any case. I've never seen a case like this before.

**Comment by mihiros6000** [comment]
I clicked Show dot files in winecfg which worked (just in case somebody comes looking). Just going to check Loader so you can close this out.

**Comment by mihiros6000** [comment]
Worked, thanks so much for your help. You're doing amazing work!
